### PR TITLE
Fixes #14635: Add support for audit mode in "apply_gm" for the ncf tests

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/14635

### CODE REVIEW

**Comment by Fdall** [comment]
Commit modified

File: tests/acceptance/default_ncf.cf.sub [hunk 1]
```diff
@@ -125,39 +128,55 @@ bundle agent apply_gm(name, args, status, result_class)
...
-      "old_class_prefix_ok" expression => "${define_expected_classes.report_string_old_class_prefix}";
-      "class_prefix_ok"     expression => "${define_expected_classes.report_string_class_prefix}";
+      "old_class_prefix_ok" expression => "${define_expected_classes.report_string_old_class_prefix_${result_class}}";
+      "class_prefix_ok"     expression => "${define_expected_classes.report_string_class_prefix_${result_class}}";
       "${result_class}_ok"  expression => "old_class_prefix_ok.class_prefix_ok",
         scope => "namespace";
 
 
   methods:
     pass1::
```
Comment (ncharles) [inline tests/acceptance/default_ncf.cf.sub]: why pass1 and not leaving the class condition empty ?

Comment (Fdall) [reply tests/acceptance/default_ncf.cf.sub]: No reasons, I just wrote it this way when trying to figure out a bug which could has come from the pass.
I will fixup this

**Comment by Fdall** [comment]
Commit modified

**Pull request merged**